From 07090b7456e14c183320dfff49573222bed82417 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Mon, 3 Nov 2014 10:52:56 +0100
Subject: [PATCH 21/21] Flag all clients2.googleclients.com

Make it possible to distinguish the different connections in wireshark.
---
 chrome/browser/component_updater/component_updater_configurator.cc  | 2 +-
 chrome/browser/media/webrtc_log_uploader.cc                         | 2 +-
 chrome/browser/net/chrome_fraudulent_certificate_reporter.cc        | 2 +-
 chrome/browser/resources/default_apps/external_extensions.json      | 4 ++--
 chrome/browser/web_resource/notification_promo.cc                   | 2 +-
 chrome/common/extensions/extension_constants.cc                     | 2 +-
 components/breakpad/app/breakpad_linux.cc                           | 2 +-
 components/breakpad/tools/crash_service.cc                          | 2 +-
 components/domain_reliability/baked_in_configs/apis_google_com.json | 2 +-
 components/domain_reliability/baked_in_configs/ddm_google_com.json  | 2 +-
 .../domain_reliability/baked_in_configs/drive_google_com.json       | 2 +-
 components/domain_reliability/baked_in_configs/mail_google_com.json | 2 +-
 components/domain_reliability/baked_in_configs/ssl_gstatic_com.json | 2 +-
 components/domain_reliability/baked_in_configs/www_google_com.json  | 2 +-
 components/domain_reliability/baked_in_configs/www_youtube_com.json | 2 +-
 components/policy/core/common/policy_loader_win.cc                  | 2 +-
 components/policy/resources/policy_templates.json                   | 6 +++---
 components/translate/core/browser/translate_url_fetcher.cc          | 1 +
 18 files changed, 21 insertions(+), 20 deletions(-)

diff --git a/chrome/browser/component_updater/component_updater_configurator.cc b/chrome/browser/component_updater/component_updater_configurator.cc
index 269dea1..2e9513f 100644
--- a/chrome/browser/component_updater/component_updater_configurator.cc
+++ b/chrome/browser/component_updater/component_updater_configurator.cc
@@ -40,7 +40,7 @@ extern const char kSwitchDisablePings[] = "disable-pings";
 const char kSwitchUrlSource[] = "url-source";
 
 #define COMPONENT_UPDATER_SERVICE_ENDPOINT \
-  "//clients2.google.com/service/update2"
+  "//trk02.clients2.google.com/service/update2"
 
 // The default url for the v3 protocol service endpoint. Can be
 // overridden with --component-updater=url-source=someurl.
diff --git a/chrome/browser/media/webrtc_log_uploader.cc b/chrome/browser/media/webrtc_log_uploader.cc
index 98b22c3..4323730 100644
--- a/chrome/browser/media/webrtc_log_uploader.cc
+++ b/chrome/browser/media/webrtc_log_uploader.cc
@@ -29,7 +29,7 @@ const int kLogCountLimit = 5;
 const uint32 kIntermediateCompressionBufferBytes = 256 * 1024;  // 256 KB
 const int kLogListLimitLines = 50;
 
-const char kUploadURL[] = "https://clients2.google.com/cr/report";
+const char kUploadURL[] = "https://trk01.clients2.google.com/cr/report";
 const char kUploadContentType[] = "multipart/form-data";
 const char kMultipartBoundary[] =
     "----**--yradnuoBgoLtrapitluMklaTelgooG--**----";
diff --git a/chrome/browser/net/chrome_fraudulent_certificate_reporter.cc b/chrome/browser/net/chrome_fraudulent_certificate_reporter.cc
index f666c87..a6b529f 100644
--- a/chrome/browser/net/chrome_fraudulent_certificate_reporter.cc
+++ b/chrome/browser/net/chrome_fraudulent_certificate_reporter.cc
@@ -24,7 +24,7 @@ namespace chrome_browser_net {
 // TODO(palmer): Switch to HTTPS when the error handling delegate is more
 // sophisticated. Ultimately we plan to attempt the report on many transports.
 static const char kFraudulentCertificateUploadEndpoint[] =
-    "http://clients3.google.com/log_cert_error";
+    "http://trk31.clients3.google.com/log_cert_error";
 
 ChromeFraudulentCertificateReporter::ChromeFraudulentCertificateReporter(
     net::URLRequestContext* request_context)
diff --git a/chrome/browser/resources/default_apps/external_extensions.json b/chrome/browser/resources/default_apps/external_extensions.json
index 52aa500..251d5cf 100644
--- a/chrome/browser/resources/default_apps/external_extensions.json
+++ b/chrome/browser/resources/default_apps/external_extensions.json
@@ -29,11 +29,11 @@
   },
   // Google Sheets
   "aapocclcgogkmnckokdopfmhonfmgoek" : {
-    "external_update_url": "https://clients2.google.com/service/update2/crx"
+    "external_update_url": "https://trk03.clients2.google.com/service/update2/crx"
   },
   // Google Slides
   "felcaaldnbdncclmgdcncolpebgiejap" : {
-    "external_update_url": "https://clients2.google.com/service/update2/crx"
+    "external_update_url": "https://trk04.clients2.google.com/service/update2/crx"
   }
 }
 
diff --git a/chrome/browser/web_resource/notification_promo.cc b/chrome/browser/web_resource/notification_promo.cc
index d6972b9..b2c12a0 100644
--- a/chrome/browser/web_resource/notification_promo.cc
+++ b/chrome/browser/web_resource/notification_promo.cc
@@ -33,7 +33,7 @@ namespace {
 
 const int kDefaultGroupSize = 100;
 
-const char promo_server_url[] = "https://clients3.google.com/crsignal/client";
+const char promo_server_url[] = "https://trk30.clients3.google.com/crsignal/client";
 
 // The name of the preference that stores the promotion object.
 const char kPrefPromoObject[] = "promo";
diff --git a/chrome/common/extensions/extension_constants.cc b/chrome/common/extensions/extension_constants.cc
index c21de09..bde4e85 100644
--- a/chrome/common/extensions/extension_constants.cc
+++ b/chrome/common/extensions/extension_constants.cc
@@ -49,7 +49,7 @@ GURL GetWebstoreSearchPageUrl(const std::string& query) {
 }
 
 const char kGalleryUpdateHttpsUrl[] =
-    "https://clients2.google.com/service/update2/crx";
+    "https://trk05.clients2.google.com/service/update2/crx";
 // TODO(battre): Delete the HTTP URL once the blacklist is downloaded via HTTPS.
 const char kExtensionBlocklistUrlPrefix[] =
     "http://www.gstatic.com/chrome/extensions/blacklist";
diff --git a/components/breakpad/app/breakpad_linux.cc b/components/breakpad/app/breakpad_linux.cc
index e80d753..00c0f3f 100644
--- a/components/breakpad/app/breakpad_linux.cc
+++ b/components/breakpad/app/breakpad_linux.cc
@@ -78,7 +78,7 @@ namespace breakpad {
 namespace {
 
 #if !defined(OS_CHROMEOS)
-const char kUploadURL[] = "https://clients2.google.com/cr/report";
+const char kUploadURL[] = "https://trk06.clients2.google.com/cr/report";
 #endif
 
 bool g_is_crash_reporter_enabled = false;
diff --git a/components/breakpad/tools/crash_service.cc b/components/breakpad/tools/crash_service.cc
index 425337f..ce262f8 100644
--- a/components/breakpad/tools/crash_service.cc
+++ b/components/breakpad/tools/crash_service.cc
@@ -24,7 +24,7 @@ namespace {
 
 const wchar_t kTestPipeName[] = L"\\\\.\\pipe\\ChromeCrashServices";
 
-const wchar_t kCrashReportURL[] = L"https://clients2.google.com/cr/report";
+const wchar_t kCrashReportURL[] = L"https://trk07.clients2.google.com/cr/report";
 const wchar_t kCheckPointFile[] = L"crash_checkpoint.txt";
 
 typedef std::map<std::wstring, std::wstring> CrashMap;
diff --git a/components/domain_reliability/baked_in_configs/apis_google_com.json b/components/domain_reliability/baked_in_configs/apis_google_com.json
index 4399439..03a750e 100644
--- a/components/domain_reliability/baked_in_configs/apis_google_com.json
+++ b/components/domain_reliability/baked_in_configs/apis_google_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "apis.google.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk08.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/ddm_google_com.json b/components/domain_reliability/baked_in_configs/ddm_google_com.json
index a14599f..3ccafab 100644
--- a/components/domain_reliability/baked_in_configs/ddm_google_com.json
+++ b/components/domain_reliability/baked_in_configs/ddm_google_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "ddm.google.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk09.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/drive_google_com.json b/components/domain_reliability/baked_in_configs/drive_google_com.json
index 0471ca7..f43dc78 100644
--- a/components/domain_reliability/baked_in_configs/drive_google_com.json
+++ b/components/domain_reliability/baked_in_configs/drive_google_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "drive.google.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk10.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/mail_google_com.json b/components/domain_reliability/baked_in_configs/mail_google_com.json
index 3fe3a48..b23f070 100644
--- a/components/domain_reliability/baked_in_configs/mail_google_com.json
+++ b/components/domain_reliability/baked_in_configs/mail_google_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "mail.google.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk11.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/ssl_gstatic_com.json b/components/domain_reliability/baked_in_configs/ssl_gstatic_com.json
index 911a0a1..1a7db08 100644
--- a/components/domain_reliability/baked_in_configs/ssl_gstatic_com.json
+++ b/components/domain_reliability/baked_in_configs/ssl_gstatic_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "ssl.gstatic.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk12.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/www_google_com.json b/components/domain_reliability/baked_in_configs/www_google_com.json
index b2129eb..3f9d3c1 100644
--- a/components/domain_reliability/baked_in_configs/www_google_com.json
+++ b/components/domain_reliability/baked_in_configs/www_google_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "www.google.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk13.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/domain_reliability/baked_in_configs/www_youtube_com.json b/components/domain_reliability/baked_in_configs/www_youtube_com.json
index ea386d7..169d473 100644
--- a/components/domain_reliability/baked_in_configs/www_youtube_com.json
+++ b/components/domain_reliability/baked_in_configs/www_youtube_com.json
@@ -4,7 +4,7 @@
   "monitored_domain": "www.youtube.com",
   "collectors": [
     {
-      "upload_url": "https://clients2.google.com/domainreliability/upload"
+      "upload_url": "https://trk14.clients2.google.com/domainreliability/upload"
     }
   ],
   "monitored_resources": [
diff --git a/components/policy/core/common/policy_loader_win.cc b/components/policy/core/common/policy_loader_win.cc
index c956a74..92c043c 100644
--- a/components/policy/core/common/policy_loader_win.cc
+++ b/components/policy/core/common/policy_loader_win.cc
@@ -68,7 +68,7 @@ const char kLegacyBrowserSupportExtensionId[] =
 
 // The web store url that is the only trusted source for extensions.
 const char kExpectedWebStoreUrl[] =
-    ";https://clients2.google.com/service/update2/crx";
+    ";https://trk15.clients2.google.com/service/update2/crx";
 // String to be prepended to each blocked entry.
 const char kBlockedExtensionPrefix[] = "[BLOCKED]";
 
diff --git a/components/policy/resources/policy_templates.json b/components/policy/resources/policy_templates.json
index 0b8bc0f..d9eaa19 100644
--- a/components/policy/resources/policy_templates.json
+++ b/components/policy/resources/policy_templates.json
@@ -1605,7 +1605,7 @@
             'dynamic_refresh': True,
             'per_profile': True,
           },
-          'example_value': ['lcncmkcnkcdbbanbjakcencbaoegdjlp;https://clients2.google.com/service/update2/crx'],
+          'example_value': ['lcncmkcnkcdbbanbjakcencbaoegdjlp;https://trk16.clients2.google.com/service/update2/crx'],
           'id': 34,
           'caption': '''Configure the list of force-installed extensions''',
           'desc': '''Allows you to specify a list of extensions that will be installed silently, without user interaction.
@@ -1614,7 +1614,7 @@
 
           For each item, <ph name="PRODUCT_NAME">$1<ex>Google Chrome</ex></ph> will retrieve the extension specified by the extension ID from the update service at the specified update URL and silently install it.
 
-          For example, <ph name="EXTENSION_POLICY_EXAMPLE">lcncmkcnkcdbbanbjakcencbaoegdjlp;https://clients2.google.com/service/update2/crx</ph> installs the <ph name="EXTENSION_POLICY_EXAMPLE_EXTENSION_NAME">Google SSL Web Search</ph> extension from the standard Chrome Web Store update URL. For more information about hosting extensions, see: <ph name="LINK_TO_EXTENSION_DOC2">http://code.google.com/chrome/extensions/hosting.html</ph>.
+          For example, <ph name="EXTENSION_POLICY_EXAMPLE">lcncmkcnkcdbbanbjakcencbaoegdjlp;https://trk17.clients2.google.com/service/update2/crx</ph> installs the <ph name="EXTENSION_POLICY_EXAMPLE_EXTENSION_NAME">Google SSL Web Search</ph> extension from the standard Chrome Web Store update URL. For more information about hosting extensions, see: <ph name="LINK_TO_EXTENSION_DOC2">http://code.google.com/chrome/extensions/hosting.html</ph>.
 
           Users will be unable to uninstall extensions that are specified by this policy. If you remove an extension from this list, then it will be automatically uninstalled by <ph name="PRODUCT_NAME">$1<ex>Google Chrome</ex></ph>. Extensions specified in this list are also automatically whitelisted for installation; the ExtensionsInstallBlacklist does not affect them.
 
@@ -4149,7 +4149,7 @@
       'features': {
         'dynamic_refresh': True,
       },
-      'example_value': [ { "extension-id": "khgabmflimjjbclkmljlpmgaleanedem", "update-url": "http://clients2.google.com/service/update2/crx" } ],
+      'example_value': [ { "extension-id": "khgabmflimjjbclkmljlpmgaleanedem", "update-url": "http://trk18.clients2.google.com/service/update2/crx" } ],
       'id': 135,
       'caption': '''List of AppPack extensions''',
       'desc': '''This policy is active in retail mode only.
diff --git a/components/translate/core/browser/translate_url_fetcher.cc b/components/translate/core/browser/translate_url_fetcher.cc
index 85988a9..db5939b 100644
--- a/components/translate/core/browser/translate_url_fetcher.cc
+++ b/components/translate/core/browser/translate_url_fetcher.cc
@@ -59,6 +59,7 @@ bool TranslateURLFetcher::Request(
   if (!extra_request_header_.empty())
     fetcher_->SetExtraRequestHeaders(extra_request_header_);
 
+  fprintf(stderr, "translator: fetching something from %s\n", url_.spec().c_str());
   fetcher_->Start();
 
   return true;
-- 
2.4.0

